packages/k8s/src/k8s/utils.ts

```diff
@@ -137,6 +137,11 @@ exec ${environmentPrefix} ${entryPoint} ${
   }
 }
 
+export function generateContainerName(image: string): string {
+  const name = image.split('/').pop()
+  return name?.split(':').at(0) || 'service'
```

Comment:
I might be wrong on this one so please correct me if I am :relaxed: 
I'd suggest throwing an exception here.
We can return 'service' only if split(':') returns first element as faulty one. That can happen in case we have an empty image, or we have some bad input like: `something/:latest`. If we return 'service' here, then the pod will fail since the image is not created well, but we can avoid the API call entirely just by throwing exception here. We know that if there is no part we can name the container, the image is faulty for sure

Reply:
Yep good point, agreed. Changed it and added some tests for it as well!